identity-apps-core/apps/authentication-portal/src/main/webapp/includes/header.jsp

```diff
@@ -29,18 +31,72 @@
 <%-- Branding Preferences --%>
 <jsp:directive.include file="branding-preferences.jsp"/>
 
-
 <%-- Extract the name of the stylesheet--%>
 <%
     String themeName = "wso2is";
+    String language = "en"; // Default language
+    Cookie[] userCookies = request.getCookies();
+    if (userCookies != null) {
+        for (Cookie cookie : userCookies) {
+            if ("ui_lang".equals(cookie.getName())) {
+                language = cookie.getValue();
+                break;
+            }
+        }
+    }
+
+    // Specify the file path
+    String filePath = application.getRealPath("/") + "/WEB-INF/classes/LanguageOptions.properties";
+
+    // Create a Map to store the language direction
+    Map<String, String> languageDirectionMap = new HashMap<>();
+
+    // Use a BufferedReader to read the file content
+    try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filePath))) {
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            // Ignore comments and empty lines
+            if (!line.trim().startsWith("#") && !line.trim().isEmpty()) {
+                // Split the line into key and value using '='
+                String[] keyValue = line.split("=");
+                if (keyValue.length == 2) { // Ensure valid format
+                    // Extract the language code from the key
+                    String[] keyParts = keyValue[0].split("\\.");
+                    String languageCode = keyParts[keyParts.length - 1];
+
+                    // Split the value into components (languageCode, languageName, direction)
+                    String[] valueParts = keyValue[1].split(",");
+                    if (valueParts.length >= 3) { // Ensure the value has at least 3 parts
+                        String direction = valueParts[2].trim();
+                        languageDirectionMap.put(languageCode, direction);
+                    } else {
+                        languageDirectionMap.put(languageCode, "ltr"); // Default to LTR if direction is missing
+                    }
+                }
+            }
+        }
+    } catch (Exception e) {
+        throw e;
+    }
+
+    // Get the current language's direction
+    String direction = languageDirectionMap.getOrDefault(language, "ltr");
+    out.write("<html dir=\"" + direction + "\">");
```

Comment:
Do we need this?

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/7220/commits/2b4f5e240882d86aa2fb38d72b2c90e7941a1a2f